SeaPublicWebsite/Views/EnergyEfficiency/AnswerSummary.cshtml

```diff
@@ -21,431 +21,672 @@
-                        {
-                            if (Model.Epc != null && (Model?.Epc.SolidWallsInsulated != null || Model?.Epc.ConstructionAgeBand != null))
-                            {
-                                <p class="govuk-body-s sea-assumed-answer govuk-!-padding-top-2">
-                                    Based on your Energy Performance Certificate, we will assume that you have @(UserDataHelper.HasInsulatedSolidWalls(Model) ? "insulated" : "uninsulated") solid walls
-                                </p>
-                            }
-                            else if (Model.YearBuilt.HasValue)
-                            {
-                                <p class="govuk-body-s sea-assumed-answer govuk-!-padding-top-1">
-                                    Based on your type of property, we will assume that you have @(UserDataHelper.HasInsulatedSolidWalls(Model) ? "insulated" : "uninsulated") solid walls
-                                </p>
-                            }
-                        }
-                    </dd>
-                    <dd class="govuk-summary-list__actions govuk-!-width-one-quarter">
-                        <a class="govuk-link" href="@Url.Action("SolidWallsInsulated_Get", "EnergyEfficiency", new {reference = Model.Reference, change = true})">
-                            Change<span class="govuk-visually-hidden"> are solid walls insulated
-                            </span>
-                        </a>
-                    </dd>
-                </div>
+                new()
+                {
+                    Href = @Url.Action(nameof(EnergyEfficiencyController.HomeAge_Get), "EnergyEfficiency", new {reference = Model.Reference, change = true}),
+                    Text = "Change",
+                    VisuallyHiddenText = "home type"
+                }
             }
-            
-            @if (Model.PropertyType != PropertyType.ApartmentFlatOrMaisonette || Model.FlatType == FlatType.GroundFloor)
-            {
-                <div class="govuk-summary-list__row">
-                    <dt class="govuk-summary-list__key govuk-!-width-one-quarter">
-                        Floor construction 
-                    </dt>
-                    <dd class="govuk-summary-list__value govuk-!-width-one-half">
-                        @(GovUkRadioCheckboxLabelTextAttribute.GetLabelText(Model.FloorConstruction))
-                        @if (Model.FloorConstruction == FloorConstruction.DoNotKnow)
+        }
+    });
+    
+    rows.Add(new()
+    {
+        Key = new() 
+        {
+            Classes = "govuk-!-width-one-quarter",
+            Text = "Wall type"
+        },
+        Value = new()
+        {
+            Classes = "govuk-!-width-one-half",
+            Html = (@<text>
+                        @(GovUkRadioCheckboxLabelTextAttribute.GetLabelText(Model.WallConstruction))
+                        @if (Model.WallConstruction == WallConstruction.DoNotKnow)
```

Comment:
Not something for you to fix now, but this logic should not be in a view file :(